src/overlays/actors/ovl_En_Stone_heishi/z_en_stone_heishi.c

```diff
@@ -35,54 +39,496 @@ const ActorInit En_Stone_heishi_InitVars = {
+}
+
+void EnStoneheishi_DrinkBottleProcess(EnStoneheishi* this, PlayState* play) {
+    f32 currentFrame = this->skelAnime.curFrame;
+    Player* player = GET_PLAYER(play);
+
+    SkelAnime_Update(&this->skelAnime);
+
+    switch (this->drinkBottleState) {
+        case EN_STONE_DRINK_BOTTLE_INITIAL:
+            if (this->timer == 0) {
+                this->textIdIndex = 4;
+                func_80151938(play, sEnStoneHeishiTextIds[this->textIdIndex]);
+                player->actor.textId = sEnStoneHeishiTextIds[this->textIdIndex];
+                this->drinkBottleState++;
+            }
+
+        default:
+            break;
+
+        case EN_STONE_DRINK_BOTTLE_DRINKING:
+            if ((Message_GetState(&play->msgCtx) == TEXT_STATE_5) && Message_ShouldAdvance(play)) {
+                Player* player = GET_PLAYER(play);
+
+                play->msgCtx.msgLength = 0;
+                player->actor.textId = 0;
+                player->exchangeItemId = PLAYER_AP_NONE;
+                this->bottleDisplay = EN_STONE_BOTTLE_RED_POTION;
+
+                if (this->playerGivesBluePotion) {
+                    this->bottleDisplay = EN_STONE_BOTTLE_BLUE_POTION;
+                }
+
+                Player_SetModels(player, 3);
+                EnStoneheishi_ChangeAnim(this, EN_STONE_HEISHI_ANIM_DRINK_1);
+                this->timer = 30;
+                this->drinkBottleState++;
+            }
+            break;
+
+        case EN_STONE_DRINK_BOTTLE_EMPTY:
+            if (this->timer != 0) {
+                if ((this->timer < 10) && (this->bottleDisplay != EN_STONE_BOTTLE_EMPTY)) {
+                    this->bottleDisplay = EN_STONE_BOTTLE_EMPTY;
+                    Actor_PlaySfxAtPos(&this->actor, NA_SE_VO_NP_DRINK);
+                    func_80123D50(play, GET_PLAYER(play), ITEM_BOTTLE, PLAYER_AP_BOTTLE);
+                }
+            } else {
+                this->drinkBottleState++;
+            }
+            break;
+
+        case EN_STONE_DRINK_BOTTLE_STAND_UP:
+            if (this->endFrame <= currentFrame) {
+                Audio_PlayFanfare(NA_BGM_GET_ITEM | 0x900);
+                this->bottleDisplay = EN_STONE_BOTTLE_NONE;
+                EnStoneheishi_ChangeAnim(this, EN_STONE_HEISHI_ANIM_STAND_UP);
+                this->drinkBottleState++;
+            }
+            break;
+
+        case EN_STONE_DRINK_BOTTLE_STANDING:
+            if (this->endFrame <= currentFrame) {
+                this->textIdIndex = 5;
+                func_80151938(play, sEnStoneHeishiTextIds[this->textIdIndex]);
+                player->actor.textId = sEnStoneHeishiTextIds[this->textIdIndex];
+                EnStoneheishi_ChangeAnim(this, EN_STONE_HEISHI_ANIM_STAND_HAND_ON_HIP);
+                this->action = EN_STONE_ACTION_1;
+                this->actionFunc = func_80BC9680;
+            }
+            break;
+    }
+}
+
+void EnStoneheishi_GiveItemPrize(EnStoneheishi* this, PlayState* play) {
```

Comment:
I'd say `Reward` over `Prize`, which sounds more like a thing you win rather than earn.